tools/gcs-bucket-mover/gcs_bucket_mover/bucket_mover_service.py

```diff
@@ -0,0 +1,657 @@
+        bucket.acl.save(acl=new_acl)
+        spinner.write('ACLs successfully copied over from the source bucket')
+
+    if source_bucket_details.default_obj_acl_entities:
+        new_default_obj_acl = _update_acl_entities(
+            config, source_bucket_details.default_obj_acl_entities)
+        bucket.default_object_acl.save(acl=new_default_obj_acl)
+        spinner.write(
+            'Default Object ACLs successfully copied over from the source bucket'
+        )
+
+    if source_bucket_details.notifications:
+        _update_notifications(spinner, config,
+                              source_bucket_details.notifications, bucket)
+        spinner.write(
+            '{} Created {} new notifications for the bucket {}'.format(
+                CHECKMARK, len(source_bucket_details.notifications),
+                bucket_name))
+
+    return bucket
+
+
+def _retry_if_false(result):
+    """Return True if we should retry because the function returned False"""
+    return result is False
+
+
+@retry(
+    retry_on_result=_retry_if_false,
+    wait_exponential_multiplier=4000,
+    wait_exponential_max=60000,
+    stop_max_attempt_number=5)
+def _create_bucket_api_call(spinner, bucket):
+    """Calls the GCS api method to create the bucket.
+
+    The method will attemp to retry up to 5 times if the 503 ServiceUnavailable
+    exception is raised.
+
+    Args:
+        spinner: The spinner displayed in the console
+        bucket: The bucket object to create
+
+    Returns:
+        True if the bucket was created, False if a ServiceUnavailable exception was raised
+
+    Raises:
+        google.cloud.exceptions.Conflict: The underlying Google Cloud api will raise this error
+            if the bucket already exists.
+    """
+
+    try:
+        bucket.create()
+    except exceptions.ServiceUnavailable:
+        spinner.write('503 Service Unavailable error returned.'
+                      ' Retrying up to 5 times with exponential backoff.')
+        return False
+    return True
+
+
+def _update_iam_policies(config, bucket, source_bucket_details):
+    """Take the existing IAM, replace the source project number with the target project
+    number and then assign the IAM to the new bucket.
+
+    Args:
+        config: A Configuration object with all of the config values needed for the script to run
+        bucket: The bucket object to update the IAM policies for
+        source_bucket_details: The details copied from the source bucket that is being moved
+    """
+
+    policy = bucket.get_iam_policy()
+
+    # Update the original policy with the etag for the policy we just got so the update is
+    # associated with our get request to make sure no other update overwrites our change
+    source_bucket_details.iam_policy.etag = policy.etag
+    for role in source_bucket_details.iam_policy:
+        members = source_bucket_details.iam_policy[role]
+        for member in members:
```

Comment:
for member in source_bucket_details.iam_policy[role]:

no need for the extra `members` variable, inlining is good.